File: Dr.NutrizioNino.Models/Dto/RecipeComparisonDto.cs
namespace Dr.NutrizioNino.Models.Dto;

public record CompareRecipeItem(
    Guid RecipeId,
    decimal QuantityGrams
);

public record CompareRecipesRequest(
    IList<CompareRecipeItem> Items
);

public record RecipeComparisonNutrientDto(
    Guid NutrientId,
    string Name,
    int PositionOrder,
    Guid UnitOfMeasureId,
    decimal Quantity
);

public record RecipeComparisonItemDto(
    Guid RecipeId,
    string Name,
    decimal BaseWeightGrams,
    decimal QuantityGrams,
    bool IsNutritionStale,
    IList<RecipeComparisonNutrientDto> Nutrients
);

public record RecipeComparisonDto(
    IList<RecipeComparisonItemDto> Items
);

File: src/Dr.NutrizioNino.WebVue/src/Interfaces/recipes/RecipeComparisonDto.ts
export interface CompareRecipeItem {
  recipeId: string
  quantityGrams: number
}

export interface CompareRecipesRequest {
  items: CompareRecipeItem[]
}

export interface RecipeComparisonNutrientDto {
  nutrientId: string
  name: string
  positionOrder: number
  unitOfMeasureId: string
  quantity: number
}

export interface RecipeComparisonItemDto {
  recipeId: string
  name: string
  baseWeightGrams: number
  quantityGrams: number
  isNutritionStale: boolean
  nutrients: RecipeComparisonNutrientDto[]
}

export interface RecipeComparisonDto {
  items: RecipeComparisonItemDto[]
}
